lua/mediaplayer/controls/dmediaplayerhtml.lua

```diff
@@ -481,4 +470,33 @@ function PANEL:MoveToCursor( xoffset, yoffset )
 	self:SetPos( cx - xoffset, cy - yoffset )
 end
 
+function PANEL:OnDocumentReady( url )
+	local console_funcs = {'log','error','debug','warn','info'}
+	for _, func in pairs(console_funcs) do
+		self:AddFunction( "console", func, function(param)
+			self:ConsoleMessage( param, func )
+		end )
+	end
+
+	self:AddFunction( "gmod", "getUrl", function( url )
+		self:SetURL( url )
+	end )
+
+
+	-- Youtube specific fix because youtube uses clientside routing which doesn't trigger OnDocumentReady and such events
+	if url == "https://www.youtube.com/" then
+		self:AddFunction( "gmod", "getUrl", function( geturl )
+			self:SetURL( geturl )
+		end )
+
+		self:QueueJavascript( [[
+			function run(){
+				if (typeof gmod == "undefined") { return };
+				gmod.getUrl(window.location.href);
+			};
+			window.addEventListener( 'yt-navigate-start', run, true )
+		]] )
+	end
```

Comment:
`FetchPageURL()` runs every second so it should pick up the URL changes through in-page navigations on YouTube. `gmod.getUrl` should still be defined since the document isn't reset.

Was there a reason why this code was needed? Depending on `yt-navigate-start` seems prone to breaking in the future.

Reply:
the issue is that YouTube uses clientside routing that makes DHTML unable to pick up when the page is loading this never triggering the code